### PR TITLE
userguide: document frame rule keyword - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata/pull/6927

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4980

Describe changes:
- Expand explanation, using description from (https://github.com/OISF/suricata/commit/a492d94826f4078d1b9299cea4f21ba3d87e3f1f). This includes explanation on the shorthand capability, as requested.

### CODE REVIEW

File: doc/userguide/rules/frame-keywords.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+The latter is useful in some cases like http, where "http" applies to
+both HTTP and HTTP/2::
+
+    alert http ... frame:http1.request;
+    alert http1 ... frame:request;
+
+This mean that these two rules are equivalent::
+
+    alert tcp ... frame:tls:pdu;
```
Comment (jasonish) [inline doc/userguide/rules/frame-keywords.rst:58]: `tls.pdu` I think.

File: doc/userguide/rules/frame-keywords.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+- pdu
+- hdr
+- data
+
+Example rules::
+
+    alert smb any any -> any any (flow:to_server; frame:smb2.pdu; content:"|FE|SMB"; startswith; sid:3;)
+    alert smb any any -> any any (flow:to_server; frame:smb2.hdr; content:"|FE|SMB"; startswith; sid:4;)
+    alert smb any any -> any any (flow:to_server; frame:smb2.data; cont:ent:"|FE|SMB"; startswith; sid:5;)
```
Comment (victorjulien) [inline doc/userguide/rules/frame-keywords.rst:92]: typo: `cont:ent`

File: doc/userguide/rules/frame-keywords.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+Usage
+-----
+
+This keyword takes an argument to specify the per protocol frame type::
+
+    alert <app proto name> ... frame:<specific frame name>
+
+Or it can specify both in the keyword::
+
+    alert tcp ... frame:<app proto name>.<specific frame name>
```
Comment (jufajardini) [inline doc/userguide/rules/frame-keywords.rst:48]: format this

File: doc/userguide/rules/frame-keywords.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+- `http`_
+- `smb`_
+- `telnet`_
+- `tls`_
+
+Examples:
+
+- tls.pdu
+- smb.smb2.hdr
+- smb.smb3.data
```
Comment (jufajardini) [inline doc/userguide/rules/frame-keywords.rst:30]: better understand how these work, before listing as examples

**Comment by jufajardini** [comment]
Feedback incorporated with: https://github.com/OISF/suricata/pull/7059

**Pull request closed without merging**